backend/cmd/cluster.go

```diff
@@ -12,10 +13,28 @@ import (
 // @todo: Use a different way to avoid name clashes with other clusters.
 const OwnClusterName = "main"
 
+var clusterAskingForAuth string
+
 type Cluster struct {
-	Name   string `json:"name"`
-	Server string `json:"server,omitempty"`
-	config *clientcmdapi.Cluster
+	Name     string `json:"name"`
+	Server   string `json:"server,omitempty"`
+	config   *clientcmdapi.Cluster
+	AuthType string `json:"auth_type"`
+}
+
+type AuthConfig struct {
+	Type         string `json:"type"`
+	ClientID     string
+	ClientSecret string
+	IdpIssuerURL string
+}
```

Comment:
I think this needs to be called `OidcConfig`, as it's only useful for that type of authentication. It could even be directly the `oauth2.Config` objects I ask to cache below.